app/parse.py

```diff
@@ -15,8 +39,99 @@ class Product:
     num_of_reviews: int
 
 
+PRODUCT_FIELDS = [field.name for field in fields(Product)]
+
+
+def normalize_text(text: str) -> str:
+    return text.replace("\xa0", " ").strip() if text else ""
+
+
+def write_products_to_csv(products: list, output_csv_path: str) -> None:
+    with open(output_csv_path + ".csv", "w", newline="") as f:
+        writer = csv.writer(f)
+        writer.writerow(PRODUCT_FIELDS)
+        writer.writerows([astuple(product) for product in products])
+
+
+def get_soup(page_source: str) -> BeautifulSoup:
+    return BeautifulSoup(page_source, "html.parser")
+
+
+def safe_get(element: Tag,
+             selector: str,
+             attr: str = None,
+             default: Any = None
+             ) -> str:
+    try:
+        selected = element.select_one(selector)
+        if not selected:
+            return default
+        return normalize_text(
+            selected[attr]
+        ) if attr else normalize_text(selected.text)
+    except (AttributeError, TypeError):
+        return default
+
+
+def get_product(product: Tag) -> Product:
+    return Product(
+        title=safe_get(product,
+                       "a.title", "title", "No Title"),
+        description=safe_get(product,
+                             "p.description", default="No Description"),
+        price=float(safe_get(product,
+                             "h4.price", default="0").replace("$", "")),
+        rating=(
+            len(product.select("span.ws-icon-star"))
+            if product.select("span.ws-icon-star")
+            else 0
+        ),
+        num_of_reviews=int(safe_get(product,
+                                    "p.review-count", default="0").split()[0]),
```

Comment:
Ensure that the element with selector `"p.review-count"` exists before attempting to split its text to avoid potential `AttributeError` or `IndexError`.